packages/venia-concept/src/components/SearchBar/searchField.js

```diff
@@ -0,0 +1,64 @@
+import React, { useCallback } from 'react';
+import { func, shape, string } from 'prop-types';
+import { useFieldState, useFormApi } from 'informed';
+import ClearIcon from 'react-feather/dist/icons/x';
+import SearchIcon from 'react-feather/dist/icons/search';
+import { useValueFromSearchParams } from '@magento/peregrine';
+
+import Icon from 'src/components/Icon';
+import TextInput from 'src/components/TextInput';
+import Trigger from 'src/components/Trigger';
+
+const clearIcon = <Icon src={ClearIcon} size={18} />;
+const searchIcon = <Icon src={SearchIcon} size={18} />;
+
+const SearchField = props => {
+    const { location, onChange, onFocus } = props;
+    const { value } = useFieldState('search_query');
+    const formApi = useFormApi();
+
+    const setValue = useCallback(
+        queryValue => {
+            // update search field
+            if (queryValue) {
+                formApi.setValue('search_query', queryValue);
+            }
+
+            // trigger the effects of clearing the field
+            if (typeof onChange === 'function') {
+                onChange('');
+            }
+        },
+        [formApi, onChange]
+    );
+
+    useValueFromSearchParams({ location, parameter: 'query', setValue });
+
+    const resetForm = useCallback(() => {
```

Comment:
This seems like something that could be built in to the API returned by `useFormApi`.

Reply:
Yes, we could probably just use `formApi.reset()` directly. I'll try that here.

Bear in mind that `useFormAPI()` is from `informed`, so we can't just add methods to it.